extensions/data-transfer/portability-data-transfer-microsoft/src/main/java/org/datatransferproject/transfer/microsoft/photos/MicrosoftPhotosImporter.java

```diff
@@ -142,143 +149,233 @@ private String createOneDriveFolder(PhotoAlbum album) throws IOException {
-      requestBuilder.header("Authorization", "Bearer " + credential.getAccessToken());
-
-      MediaType contentType = MediaType.parse(photo.getMediaType());
-
-      StreamingBody body = new StreamingBody(contentType, inputStream);
-
-      // PUT the stream
-      requestBuilder.put(body);
-
-      try (Response response = client.newCall(requestBuilder.build()).execute()) {
-        int code = response.code();
-        ResponseBody responseBody = response.body();
-        if (code == 401){
-            // If there was an unauthorized error, then try refreshing the creds
-            credentialFactory.refreshCredential(credential);
-            monitor.info(() -> "Refreshed authorization token successfuly");
-
-            requestBuilder.header("Authorization", "Bearer " + credential.getAccessToken());
-            Response newResponse = client.newCall(requestBuilder.build()).execute();
-            code = newResponse.code();
-            responseBody = newResponse.body();
-        }
-        if (code < 200 || code > 299) {
-          throw new IOException(
-              "Got error code: " + code + " message: " + response.message() + " body: " + response
-                  .body().string());
-        }
-
-        // Extract photo ID from response body
-        Preconditions.checkState(body != null, "Got Null Body when creating photo %s", photo);
-        Map<String, Object> responseData = objectMapper.readValue(responseBody.bytes(), Map.class);
-        String photoId = (String) responseData.get("id");
-        checkState(!Strings.isNullOrEmpty(photoId),
-            "Expected id value to be present in %s", responseData);
-        return photoId;
+    // upload file in 32000KiB chunks
+    int CHUNK_SIZE = 32000 * 1024; // 32000KiB
+
+    // Arrange the data to be uploaded in chunks
+    ArrayList<Chunk> chunksToSend = new ArrayList();
+    byte[] data = new byte[CHUNK_SIZE];
+    int totalFileSize = 0;
+    int quantityToSend;
+    int roomLeft = CHUNK_SIZE;
+    int offset = 0;
+    int chunksRead = 0;
+
+
+    // start timing
+    Instant start = Instant.now();
+    while ((quantityToSend = inputStream.read(data, offset, roomLeft)) != -1) {
+      offset += quantityToSend;
+      roomLeft -= quantityToSend;
+      if (roomLeft == 0) {
+        chunksToSend.add(new Chunk(data, CHUNK_SIZE, chunksRead * CHUNK_SIZE));
+        chunksRead++;
+        roomLeft = CHUNK_SIZE;
+        offset = 0;
+        totalFileSize += CHUNK_SIZE;
+        data = new byte[CHUNK_SIZE];
       }
-    } finally {
-      if (inputStream != null) {
-        try {
-          inputStream.close();
-        } catch (IOException e1) {
-          monitor.info(() -> "Couldn't close input stream");
-        }
+    }
+    if (offset != 0) {
+      chunksToSend.add(new Chunk(data, offset, chunksRead * CHUNK_SIZE));
+      totalFileSize += offset;
+      chunksRead++;
+    }
+    final int _debugTotalFileSize = totalFileSize;
```

Comment:
do you need these? it seems like you only use them once without modifying them before that